### PR TITLE
[extension][ecsobserver] Init config and target definition

### PR DESCRIPTION
**Pull request by pingleig**
**Description:** <Describe what has changed.>

A smaller part of https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2734
Init config and some struct definitions for ECS Prometheus service discovery

btw: 1.4k loc are go.sum, it only have 500 lines for config struct and some large struct for exporting.

```
===============================================================================
 Language            Files        Lines         Code     Comments       Blanks
===============================================================================
 Go                     17          961          562          288          111
 Makefile                1            1            1            0            0
 YAML                    1           40           37            1            2
```

**Link to tracking Issue:** <Issue number if applicable>

- Feature request #1395 
- Config PR (merged) https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2377
- Previous PR (that is too big) #2734 

**Testing:**

unit test

**Documentation:** <Describe the documentation added.>

No new doc, see https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2377 for existing doc

### CODE REVIEW

**Review by mxiamxia — APPROVED** [review]
LGTM!

**Comment by pingleig** [comment]
ping @jrcamp

File: extension/observer/ecsobserver/config.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ecsobserver
+
+import (
+	"os"
+	"time"
+
+	"go.opentelemetry.io/collector/config/configmodels"
```
Comment (jrcamp) [inline extension/observer/ecsobserver/config.go:21]: config models has been moved around recently, need to rebase.

**Review by jrcamp — COMMENTED** [review]
Thanks for splitting up. Just needs rebasing then lgtm.

File: extension/observer/ecsobserver/config.go (hunk 1 above)
Comment (pingleig) [reply extension/observer/ecsobserver/config.go]: Rebased and changed to `.../config`

**Comment by pingleig** [comment]
@jrcamp rebased and also bumped aws sdk version to match other components.

**Review by jrcamp — APPROVED** [review]

**Comment by pingleig** [comment]
ping @bogdandrutu

**Pull request merged**